lib/Server.js

```diff
@@ -744,36 +744,46 @@ class Server {
     }
 
     if (this.options.open) {
-      const openTarget = prettyPrintUrl(this.hostname || 'localhost');
+      const openTarget = prettyPrintUrl(this.options.host || 'localhost');
 
       runOpen(openTarget, this.options.open, this.logger);
     }
   }
 
-  listen(port, hostname, fn) {
-    if (hostname === 'local-ip') {
-      this.hostname = internalIp.v4.sync() || internalIp.v6.sync() || '0.0.0.0';
-    } else if (hostname === 'local-ipv4') {
-      this.hostname = internalIp.v4.sync() || '0.0.0.0';
-    } else if (hostname === 'local-ipv6') {
-      this.hostname = internalIp.v6.sync() || '::';
-    } else {
-      this.hostname = hostname;
+  listen(port, hostname = this.options.host, fn) {
```

Comment:
Why we need default assign? It can be some misleading and we may use it by mistake like we do with port before, I prefer to use always  `this.options.host` and `this.options.port`

Reply:
> const devServer = new DevServer({ host: '192.168.0.2' }); devServer.listen(80); should work and listen 192.168.0.2 

Else this case will throw error too.

Reply:
I think we can refactor it to avoid using `hostname`, I mean we need simplify logic here, code is good

Reply:
Hmm, we can do `listen(port, fn)`

Reply:
We will do it late, just assign `hostname` to `this.options.host` after check